server/src/lsp-server.ts

```diff
@@ -93,91 +93,94 @@ export class LspServer {
         return bundled;
     }
 
-    async initialize(params: TypeScriptInitializeParams): Promise<TypeScriptInitializeResult> {
-        this.logger.log('initialize', params);
-        this.initializeParams = params;
-
-        const { logVerbosity, plugins }: TypeScriptInitializationOptions = {
-            logVerbosity: this.options.tsserverLogVerbosity,
-            plugins: [],
-            ...this.initializeParams.initializationOptions
-        };
-        const logFile = this.getLogFile(logVerbosity);
-        const globalPlugins: string[] = [];
-        const pluginProbeLocations: string[] = [];
-        for (const plugin of plugins) {
-            globalPlugins.push(plugin.name);
-            pluginProbeLocations.push(plugin.location);
-        }
-
-        const tsserverPath = this.findTsserverPath();
-        this.tspClient = new TspClient({
-            tsserverPath,
-            logFile,
-            logVerbosity,
-            globalPlugins,
-            pluginProbeLocations,
-            logger: this.options.logger,
-            onEvent: this.onTsEvent.bind(this)
-        });
-
-        this.tspClient.start();
-        this.tspClient.request(CommandTypes.Configure, {
-            preferences: {
-                allowTextChangesInNewFiles: true
+    initialize(preferences?: tsp.UserPreferences) {
+        return async function(params: TypeScriptInitializeParams): Promise<TypeScriptInitializeResult> {
```

Comment:
There's probably a better way to do this, but for now I just curried the initialize function to pass down the user preferences.